tests/test_association_db_sqlite.py

```diff
@@ -0,0 +1,321 @@
+            scatter_arcsec=0.0)
+        if not update_spatial_index:
+            # Set the spatial index to some arbitray value
+            dia_objects[0].dia_object_record.set('indexer_id', 10)
+        dia_collection = DIAObjectCollection(dia_objects)
+        self.assoc_db = AssociationDBSqliteTask()
+        self.assoc_db.create_tables()
+
+        self.assoc_db.store(dia_collection, update_spatial_index)
+
+        self.assoc_db._db_cursor.execute(
+            "SELECT * FROM dia_objects")
+        for row in self.assoc_db._db_cursor.fetchall():
+            if update_spatial_index:
+                # Index is HTM cell number at level=7, RA,DEC=0.1
+                dia_objects[0].dia_object_record.set('indexer_id', 253952)
+            round_trip_object = \
+                self.assoc_db._dia_object_converter.source_record_from_db_row(
+                    row)
+            self._compare_source_records(
+                round_trip_object,
+                dia_objects[0].dia_object_record)
+
+    def test_store_updated(self):
+        """ Test the storage of newly associated DIAObjects and DIASources.
+        """
+        dia_objects = create_test_dia_objects(n_objects=1, n_sources=1)
+        dia_collection = DIAObjectCollection(dia_objects)
+        self.assoc_db = AssociationDBSqliteTask()
+        self.assoc_db.create_tables()
+
+        self.assoc_db.store(dia_collection, True)
+        self.assoc_db._commit()
+
+        new_dia_object = create_test_dia_objects(
+            n_objects=1, n_sources=1, start_id=1)
+        dia_collection.append(new_dia_object[0])
+        dia_collection.update_dia_objects()
+        dia_collection.update_spatial_tree()
+
+        self.assoc_db.store_updated(dia_collection, [1])
+
+    def test_get_dia_objects(self):
+        """ Test the retrieval of DIAObjects from the database.
+        """
+        dia_objects = create_test_dia_objects(
+            n_objects=2, n_sources=2, scatter_arcsec=0.0)
+        dia_collection = DIAObjectCollection(dia_objects)
+        self.assoc_db = AssociationDBSqliteTask()
+        self.assoc_db.create_tables()
+        self.assoc_db.store(dia_collection, True)
+
+        self.assoc_db._commit()
+
+        self.assoc_db._db_cursor.execute(
+            "SELECT indexer_id FROM dia_objects")
+        indexer_ids = np.array(
+            self.assoc_db._db_cursor.fetchall(), np.int).flatten()
+
+        output_dia_objects = self.assoc_db.get_dia_objects(indexer_ids)
+
+        for obj_idx in range(2):
+            self.assertEqual(
+                output_dia_objects[obj_idx].n_dia_sources, 2)
+            self._compare_source_records(
+                output_dia_objects[obj_idx].dia_object_record,
+                dia_collection.dia_objects[obj_idx].dia_object_record)
+
+            for record_a, record_b in zip(
+                    output_dia_objects[obj_idx].dia_source_catalog,
+                    dia_collection.dia_objects[obj_idx].dia_source_catalog):
```

Comment:
Again, can you safely assume order is preserved?

Reply:
Change this to a similar sort on id as before.